tests/unit/actor/test_configuration.py

```diff
@@ -16,6 +27,204 @@ def test_disable_browser_sandbox(
+        await crawler_2.add_requests(
+            [Request.from_url(url='http://example.com/2'), Request.from_url(url='http://example.com/3')]
+        )
+
+        await crawler_1.run()
+        await crawler_2.run()
+
+    assert service_locator.get_configuration() is config_actor
+    assert crawler_1._service_locator.get_configuration() is config_crawler_1
+    assert crawler_2._service_locator.get_configuration() is config_crawler_2
+
+    assert crawler_1.statistics.state.requests_total == 1
+    assert crawler_2.statistics.state.requests_total == 2
+
+
+async def test_crawler_global_configuration() -> None:
+    """Test that crawler and Actor use service_locator based configuration unless explicit configuration
+    was passed to them."""
+    config_global = ApifyConfiguration()
+    service_locator.set_configuration(config_global)
+
+    async with Actor():
+        crawler = BasicCrawler()
+
+    assert service_locator.get_configuration() is config_global
+    assert crawler._service_locator.get_configuration() is config_global
+
+
+async def test_crawler_uses_implicit_apify_config() -> None:
+    """Test that Actor is using implicit ApifyConfiguration in Actor context."""
+    async with Actor:
+        assert isinstance(Actor.config, ApifyConfiguration)
+
+
+async def test_storage_retrieved_is_different_with_different_config(tmp_path: Path) -> None:
+    """Test that retrieving storage depends on used configuration."""
+    dir_1 = tmp_path / 'dir_1'
+    dir_2 = tmp_path / 'dir_2'
+    config_actor = ApifyConfiguration()
+    config_actor.storage_dir = str(dir_1)
+    config_crawler = ApifyConfiguration()
+    config_crawler.storage_dir = str(dir_2)
+
+    async with Actor(configuration=config_actor):
+        actor_kvs = await Actor.open_key_value_store()
+        crawler = BasicCrawler(configuration=config_crawler)
+        crawler_kvs = await crawler.get_key_value_store()
+
+    assert actor_kvs is not crawler_kvs
+
+
+async def test_storage_retrieved_is_same_with_equivalent_config() -> None:
+    """Test that retrieving storage depends on used configuration. If two equivalent configuration(even if they are
+    different instances) are used it returns same storage."""
+    config_actor = ApifyConfiguration()
+    config_crawler = ApifyConfiguration()
+
+    async with Actor(configuration=config_actor):
+        actor_kvs = await Actor.open_key_value_store()
+        crawler = BasicCrawler(configuration=config_crawler)
+        crawler_kvs = await crawler.get_key_value_store()
+
+    assert actor_kvs is crawler_kvs
+
+
+async def test_storage_retrieved_is_same_with_same_config() -> None:
+    """Test that retrieving storage is same if same configuration is used."""
+    async with Actor():
+        actor_kvs = await Actor.open_key_value_store()
+        crawler = BasicCrawler()
+        crawler_kvs = await crawler.get_key_value_store()
+
+    assert actor_kvs is crawler_kvs
```

Comment:
This is horrible that we have `get_...` and `open_...` variants ... couldn't we unify it? I would use `open_...` in Crawlee as well.

Reply:
Sure, and we will probably want to do the changes in the Crawlee (rather open than get).

Reply:
https://github.com/apify/crawlee-python/issues/1415